### PR TITLE
feat: Make https secretclass configurable

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Fixes https://github.com/stackabletech/nifi-operator/issues/499

Alternate take on https://github.com/stackabletech/nifi-operator/pull/529, but a little bit more simplistic.
I also renamed `clusterConfig.tls.httpSecretClass` to `clusterConfig.tls.serverSecretClass` to be consistent with other operators (like zk or trino).

I skipped the whole "whether to request node certs" thing, as listener-op will replace it anyway.

## CRD change

```yaml
spec:
  clusterConfig:
    tls: # mandatory
      serverSecretClass: non-default-secret-class # mandatory, defaults to tls
```

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [x] CRD changes approved
- [x] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [x] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

**Comment by sbernauer** [comment]
We talked about this in daily and took a decision shortcut as this is mostly a copy/paste from trino-operator.

File: rust/crd/src/tls.rs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+use serde::{Deserialize, Serialize};
+use stackable_operator::schemars::{self, JsonSchema};
+
+#[derive(Clone, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NifiTls {
+    /// Only affects client connections.
+    /// This setting controls:
+    /// - Which cert the servers should use to authenticate themselves against the client
```
Comment (NickLarsenNZ) [inline rust/crd/src/tls.rs:9]: The list of one item seems redundant.

Also, is this statement true? 

> ... to authenticate themselves against the client

```suggestion
    /// This only affects client connections and is used to
    /// control which certificate the servers should use to
    /// authenticate themselves against the client.
```

**Review by NickLarsenNZ — APPROVED** [review]
LGTM, one nit

File: rust/crd/src/tls.rs (hunk 1 above)
Comment (sbernauer) [reply rust/crd/src/tls.rs:9]: I know the list is redundant, I just wanted to be consistent with other operators, such as https://github.com/stackabletech/trino-operator/blob/9d5b2145efd3938790dd2ff532c1437d282d6ef0/rust/crd/src/lib.rs#L278.

But I guess the typical user wont notice, so just using you suggestion

**Review by NickLarsenNZ — APPROVED** [review]
LGTM

**Pull request merged**